cnxauthoring/storage/postgresql.py

```diff
@@ -328,11 +329,20 @@ def update(self, item_or_items):
 
     def persist(self):
         """Persist/commit the changes."""
-        self.conn.commit()
+        # A try/except statement had been added to handle
+        # situations were the database connection is lost but not caught before
+        # a call to this method (see unittest test_functional.
+        # FunctionalTests.test_storage_unavailable_error). Would suggest
+        # moving all error catching inside this class in the future.
+        try:
+            self.conn.commit()
+        except self.Error:
+            raise httpexceptions.HTTPServiceUnavailable
```

Comment:
Actually, views.py post_content has probably changed since that issue was raised.  Now if `storage.persist()` fails, it'll raise an exception (storage.Error) which is also not acceptable...

The code (line 345) is currently like this
```python
try:
    for r in resources:
        storage.add(r)
except ArchiveConnectionError:
    raise httpexceptions.HTTPBadRequest(
        'Derived failed: {}'.format(derived_from))
except storage.Error:
    storage.abort()
else:
    storage.persist()
```

I think it probably should be:
```python
try:
    for r in resources:
        storage.add(r)
    storage.persist()
except ArchiveConnectionError:
    raise httpexceptions.HTTPBadRequest(
        'Derived failed: {}'.format(derived_from))
except storage.Error:
    storage.abort()
```